server/services/cohorts.js

```diff
@@ -100,6 +108,80 @@ const getCountOfAllocation = async ({ cohortId } = {}) =>
     cohort_id: cohortId,
   });
 
+const changeCohortParticipant = async (
+  { cohortId, participantId, newCohortId, meta } = { meta: {} }
+) => {
+  // Get existing participant cohort map
+  const participantCohort = await dbClient.db[collections.COHORT_PARTICIPANTS].findOne({
+    participant_id: participantId,
+  });
+  if (!participantCohort || !participantCohort.id) {
+    throw new Error('Participant not found in cohort');
+  }
+  // Get Graduation Status of Participant
+  const participantPostHireStatuses =
+    (await getPostHireStatusesForParticipant({ participantId })) || [];
+  // Get cohort details for both
+  const newCohort = await dbClient.db[collections.COHORTS].findOne({
+    id: newCohortId,
+  });
+
+  // Update all data in transaction
+  const resp = await dbClient.db.withTransaction(async (tnx) => {
+    // Update Cohort Participant
+    await tnx[collections.COHORT_PARTICIPANTS].update(participantCohort.id, {
+      cohort_id: newCohortId,
+    });
+    // Update post-hire status
+    let newPostHireStatuses;
+    if (participantPostHireStatuses.length > 0) {
+      // Delete all post-hire graduations statuses
+      // Delete Post-Hire Status
+      await tnx[collections.PARTICIPANT_POST_HIRE_STATUS].destroy({
+        participant_id: participantId,
+        status: postHireStatuses.postSecondaryEducationCompleted,
+      });
+    }
+    // Now check New cohort is expired or not
+    if (newCohort.end_date && newCohort.end_date < new Date()) {
```

Comment:
Isn't it safe to throw an exception when trying to change to an ongoing cohort?

Reply:
Throwing exception before transaction .... like extended validation checj